### ISSUE TITLE
Installation Error

### ISSUE THREAD

**Issue by ajhenley** [openedx-unsupported/configuration#1640]
Using either the manual installation or the one-step installation found here: https://github.com/edx/configuration/wiki/edX-Ubuntu-12.04-64-bit-Installation, I keep getting this error:

TASK: [common | Add git apt repository] **************************************\* 
failed: [localhost] => {"failed": true, "item": ""}
msg: Failed to validate the SSL certificate for launchpad.net:443. Use validate_certs=no or make sure your managed systems have a valid CA certificate installed. Paths checked for this platform: /etc/ssl/certs, /etc/pki/ca-trust/extracted/pem, /etc/pki/tls/certs, /usr/share/ca-certificates/cacert.org, /etc/ansible

FATAL: all hosts have already failed -- aborting

Any idea what is going on here?

**Comment by diaspar** [comment]
This exact error is happening to me too.

**Comment by disisid** [comment]
Same issue. Using valid_certs=no also doesn't help.

**Comment by diaspar** [comment]
This error happens in a fresh digital ocean droplet with ubuntu 12.04

Just in case I ran: /usr/sbin/update-ca-certificates

and I manually verified /etc/ssl/certs, and its has the normal set of certs.

The error persists.

**Comment by disisid** [comment]
Raised the issue here: https://groups.google.com/forum/#!topic/edx-code/sR-6VhLjcPU

**Comment by ajhenley** [comment]
hmmm...I am also using digital ocean, but just a week ago I was able to install without issue...

**Comment by singingwolfboy** [comment]
Based on the error message, it appears that the problem is with launchpad.net, which is outside of edX's control. It looks like the failing task is [in common/tasks/main.yml](https://github.com/edx/configuration/blob/d56d560765e5c9674e698e2566abd0e8fb526787/playbooks/roles/common/tasks/main.yml#L35-L36) -- @disisid, when you say that using `valid_certs=no` didn't help, did you apply it to this task, or did you put it somewhere else?

It looks like [the last commit in the repo to modify this file](https://github.com/edx/configuration/commit/df371fa48fe4c00dccfdae47bd52959e7790c86a) was made two days ago, with the message "Update CA Certificates to prevent SSL errors." Ironic, since this appears to be a SSL error. @fredsmith, do you know anything about this? Could this commit be related to the problem we're seeing?

**Comment by davefung** [comment]
Don't know where should I place "valid_certs=no", and it still failed after I inputed "sudo /usr/sbin/update-ca-certificates". Thanks.

**Comment by fredsmith** [comment]
https://github.com/edx/configuration/pull/1641 should fix this.  the issue is an ansible bug due to the way that ansible concatenates CA certificates.  _sometimes_ update-ca-certs fixes it by shuffling the order of certificates, but generally it's broken.

the bug has supposedly been fixed by https://github.com/ansible/ansible/pull/7236, but that change is not in the version of ansible we support.

**Comment by diaspar** [comment]
@fredsmith thanks. I was using trying to use this command to install edx: `wget https://raw.githubusercontent.com/edx/configuration/master/util/install/vagrant.sh -O - | bash`

It is still not working. I suppose your PR has to be merged first.

**Comment by ajhenley** [comment]
using the update command still gives the same error as well...

I have also tried to install friom scratch again like diaspar and ran into the same thing.

**Comment by fredsmith** [comment]
@diaspar @ajhenley - my change has been merged.  Please try again after updating.

**Comment by ajhenley** [comment]
@fredsmith : Sorry if this is a stupid question, but after updating what? the command is pulling from the repository, shouldn't it be grabbing the most current code?

**Comment by diaspar** [comment]
@fredsmith same question as @ajhenley

**Comment by fredsmith** [comment]
we're still working on this - hold on.

**Comment by fredsmith** [comment]
ok - we merged in a more complete fix and tested this.  re-running the vagrant script should work this time.  let me know otherwise.

**Comment by mlkwaqas** [comment]
@fredsmith I'm still getting the error

**Comment by diaspar** [comment]
@fredsmith so do I. Is there something we have to erase before running again: `https://raw.githubusercontent.com/edx/configuration/master/util/install/vagrant.sh -O - | bash`

**Comment by carherbl** [comment]
same here. Tried on a fresh installation and still not working

**Comment by fredsmith** [comment]
the vagrant script was pulling the release branch of configuration. this change has now been merged into release.  please try again.

**Comment by carherbl** [comment]
still got this error. Dunno if I have to start on a clean installation or not. 

TASK: [common | Add git apt repository] ***********************************
failed: [localhost] => {"failed": true, "item": ""}
msg: Failed to validate the SSL certificate for launchpad.net:443. Use vali                                                                                          certs=no or make sure your managed systems have a valid CA certificate inst                                                                                          . Paths checked for this platform: /etc/ssl/certs, /etc/pki/ca-trust/extrac                                                                                          em, /etc/pki/tls/certs, /usr/share/ca-certificates/cacert.org, /etc/ansible

FATAL: all hosts have already failed -- aborting

**Comment by diaspar** [comment]
I had to manually erase /var/tmp/configuration  (configuration repo)

now it started, and the error seems gone. Will report soon.

**Comment by carherbl** [comment]
Ok, done on a clean ubuntu installation. Now it works. Thanks!

**Comment by disisid** [comment]
sudo /edx/bin/update configuration master

failing at
fatal: [localhost] => error while evaluating conditional: ansible_distribution == 'Ubuntu'

**Comment by singingwolfboy** [comment]
@disisid Can you confirm that you're using ansible 1.5.5, [as specified in the requirements.txt file](https://github.com/edx/configuration/blob/master/requirements.txt#L1)?

**Comment by disisid** [comment]
@singingwolfboy  yes 
root@ip-172-31-2-100:/var/tmp# ansible --version
ansible 1.5.5

**Comment by diaspar** [comment]
works for me, thanks!

**Comment by fredsmith** [comment]
@disisid - can you paste a bit more of that error message?  I need to know what playbook it's executing.  whats the line before that that says "ansible-playbook /something/.yml" ?

**Comment by TangXT** [comment]
I deleted /var/tmp/configuration
and then run
cd /var/tmp
git clone -b release https://github.com/edx/configuration

cd /var/tmp/configuration
sudo pip install -r requirements.txt

Then run 

cd /var/tmp/configuration/playbooks && sudo ansible-playbook -c local ./edx_sandbox.yml -i "localhost,"

It fixed!

**Comment by disisid** [comment]
@fredsmith I had already terminated the server and spun a new server where it worked fine. Thanks anyways

**Comment by luiscleto** [comment]
I also get a similar error:
"failed: [localhost] => {"failed": true, "item": ""}
msg: Failed to validate the SSL certificate for download.elasticsearch.org:443. Use validate_certs=no or make sure your managed systems have a valid CA certificate installed. Paths checked for this platform: /etc/ssl/certs, /etc/pki/ca-trust/extracted/pem, /etc/pki/tls/certs, /usr/share/ca-certificates/cacert.org, /etc/ansible

FATAL: all hosts have already failed -- aborting"

It happens whenever I try to install the full stack using:
https://raw.githubusercontent.com/edx/configuration/master/util/install/vagrant.sh

I'm using a digitalocean droplet. I've tried deleting /var/tmp/configuration and cloning the release but I got the same error.

**Comment by grillonbleu** [comment]
I have the exact same error as @luiscleto. I'm trying to install the full stack edX using:
https://raw.githubusercontent.com/edx/configuration/master/util/install/vagrant.sh

I'm using a clean Ubuntu Server 12.04.5 LTS install on a VirtualBox virtual machine.

**Comment by ovnicraft** [comment]
I got them same donwloading logstash:
Check this: https://github.com/edx/configuration/blob/named-release/birch/playbooks/roles/logstash/defaults/main.yml#L9

**Comment by wwj718** [comment]
@luiscleto @lampyridae 
Upgrade ansible version to 1.7 . it worked fine :smile:

**Comment by luiscleto** [comment]
@wwj718 That made it crash in a whole different way trying to "update the ssh motd on Ubuntu" due to some inexistent conf file (/etc/update-motd.d/51-cloudguest)

**Comment by e0d** [comment]
What version of ubuntu?

Ed

**Comment by luiscleto** [comment]
12.04.05 x64

**Comment by e0d** [comment]
The issue I believe is that there's a reference to an aws specific motd
file in the common role, common/tasks/main.yml:127.  This should be moved
to the aws role.  I installed Aspen on a droplet some time back, but not
Birch.  You could start by moving that item out of common.  PRs welcome.

Ed
(e0d on github)

**Comment by luiscleto** [comment]
Well, I moved the item from common to aws but now I'm getting a python error in task "edxapp | syncdb and migrate". All I've changed so other than that was the ansible version to 1.7.

This is the logfile I had ansible generate: https://gist.github.com/luiscleto/08fd2abad5c1794c3176

Edit: Although I was now using the master branch of the configuration repo instead of the release branch.

**Comment by jp-bpl** [comment]
I've created a PR for the issue with "download.elasticsearch.org:443" : https://github.com/edx/configuration/pull/1912
This could be applied to where ever this issue appears, by simply ignoring certificates.

**Comment by luiscleto** [comment]
I've applied that change as well and managed to install it yesterday but it's more of a workaround than a fix.
I tried upgrading the ansible version but every time I fixed an issue and got to the next part of the installation, there would be another issue there. But at least there were no issues with the certificates, just with the yaml parsing of jinja variables that then caused a bunch of python errors. Plus a few tasks were where they shouldn't be, among a few other things. Either way, I paused work on my configuration and just forced it to not validate the certificate. Maybe I'll try again some other time.

**Comment by VincentXWD** [comment]
failed: [localhost] => {"failed": true, "item": ""}
msg: Failed to validate the SSL certificate for download.elasticsearch.org:443. Use validate_certs=no or make sure your managed systems have a valid CA certificate installed. Paths checked for this platform: /etc/ssl/certs, /etc/pki/ca-trust/extracted/pem, /etc/pki/tls/certs, /usr/share/ca-certificates/cacert.org, /etc/ansible

FATAL: all hosts have already failed -- aborting

I was stuck here for about several days. Hope for someone can solve it and help me.

**Comment by jp-bpl** [comment]
@misakaimoto the pr #1912 has been merged, if you pull the latest you should be fine.

**Comment by VincentXWD** [comment]
Thanks very much. I'm going to have a try now. :)

**Comment by VincentXWD** [comment]
Yes, it does work! Thanks once again.
